infrastructure/src/main/java/org/corfudb/infrastructure/BaseServer.java

```diff
@@ -54,73 +79,70 @@ public void shutdown() {
+    public void handlePing(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
+        log.debug("handlePing[{}]: Ping message received from {} {}", req.getHeader().getRequestId(),
+                req.getHeader().getClientId().getMsb(), req.getHeader().getClientId().getLsb());
+
+        HeaderMsg responseHeader = getHeaderMsg(req.getHeader(), false, true);
+        ResponseMsg response = getResponseMsg(responseHeader, getPingResponseMsg());
+        r.sendResponse(response, ctx);
     }
 
     /**
      * Respond to a version request message.
      *
-     * @param msg   The incoming message
+     * @param req   The incoming message
      * @param ctx   The channel context
      * @param r     The server router.
      */
-    @ServerHandler(type = CorfuMsgType.VERSION_REQUEST)
-    private void getVersion(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        VersionInfo vi = new VersionInfo(serverContext.getServerConfig(),
-                                         serverContext.getNodeIdBase64());
-        r.sendResponse(ctx, msg, new JSONPayloadMsg<>(vi, CorfuMsgType.VERSION_RESPONSE));
+    @RequestHandler(type = RequestPayloadMsg.PayloadCase.VERSION_REQUEST)
+    private void getVersion(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
+        VersionInfo versionInfo = new VersionInfo(serverContext.getServerConfig(), serverContext.getNodeIdBase64());
+        HeaderMsg responseHeader = getHeaderMsg(req.getHeader(), false, true);
+        ResponseMsg response = getResponseMsg(responseHeader, getVersionResponseMsg(versionInfo));
+        r.sendResponse(response, ctx);
     }
 
     /**
-     * Respond to a epoch change message.
+     * Respond to a epoch change request.
      * This method also executes sealing logic on each individual server type.
      *
-     * @param msg The incoming message
-     * @param ctx The channel context
-     * @param r   The server router.
+     * @param req The incoming request message.
+     * @param ctx The channel context.
+     * @param r The server router.
      */
-    @ServerHandler(type = CorfuMsgType.SEAL)
-    public synchronized void handleMessageSetEpoch(@NonNull CorfuPayloadMsg<Long> msg,
-                                                   ChannelHandlerContext ctx,
-                                                   @NonNull IServerRouter r) {
+    @RequestHandler(type = RequestPayloadMsg.PayloadCase.SEAL_REQUEST)
+    private synchronized void handleSeal(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
         try {
-            long epoch = msg.getPayload();
+            final long epoch = req.getPayload().getSealRequest().getEpoch();
             String remoteHostAddress;
             try {
                 remoteHostAddress = ((InetSocketAddress)ctx.channel().remoteAddress()).getAddress().getHostAddress();
-            } catch (NullPointerException e) {
+            } catch (NullPointerException ex) {
```

Comment:
Comments unaddressed from previous PR. don't think you need a try-catch here.

Reply:
It's at least convenient for testing. The unit tests that do not use mocking fail because of this. See LayoutSealTest, for example.